### ISSUE TITLE
npm install issues

### ISSUE THREAD

**Issue by DaveLindberg** [npm/npm#7376]
running npm install -g grunt-cli
results include:
```
npm WARN package.json resolve@0.3.1 No readme data.
npm WARN package.json github-url-from-git@1.1.1 No repository field.
npm WARN package.json assert-plus@0.1.2 No repository field.
npm WARN package.json ctype@0.5.2 No repository field.
```
...and multiple errors on npm install -g bower. A number of node_modules failed during install. Almost all are Error code ENOENT, errno 34. 
Here's a partial list:
```
npm ERR! Error: ENOENT, lstat '/Users/xxx/local/lib/node_modules/bower/node_modules/mout/lang.js'
.../tmp/test/name-test.js'
.../fstream/lib/file-writer.js'
.../cardinal/lib/highlightFile.js'
.../fstream-ignore/test/.ignore'
.../bower-json/test/pkg-bower-json-malformed/bower.json'
```
and so on, for maybe 10 +/- similar messages.

running Mac OSX Yosemite 10.10.2

**Comment by KenanY** [comment]
@DaveLindberg If a `npm-debug.log` file is created, could you upload it as a [gist](https://gist.github.com/)? Posting small portions of the log doesn't reveal the issue.

**Comment by othiym23** [comment]
Additionally, @DaveLindberg, you don't say what version of npm you're running. If it's older than `npm@2.1.5`, you should upgrade to the latest stable version using `npm install -g npm` (prefixing with `sudo` if you originally installed node and / or npm as root). There were many race conditions in older versions of npm that have since been addressed, and which manifested as the `lstat ENOENT` issues you describe.

Also:

```
npm WARN package.json resolve@0.3.1 No readme data.
npm WARN package.json github-url-from-git@1.1.1 No repository field.
npm WARN package.json assert-plus@0.1.2 No repository field.
npm WARN package.json ctype@0.5.2 No repository field.
```

These warnings may be safely ignored, as they are telling you that various helpful fields are missing from the `package.json` fields of dependencies. These warnings are of debatable utility for you as an end user and will probably be going away soon as a result.

**Comment by DaveLindberg** [comment]
Thanks @othiym23 for the update info. I currently have npm version 1.2.28. Running the upgrade command produced a slew of unmet dependencies. See the gist at https://gist.github.com/DaveLindberg/dcc75fc10e7008728356

**Comment by DaveLindberg** [comment]
@KenanY Thanks for the help! I've created a gist from the stdout, available at https://gist.github.com/DaveLindberg/b89875b8918f614cea84

**Comment by othiym23** [comment]
@DaveLindberg You will have a better experience if you [upgrade](https://github.com/npm/npm/wiki/Troubleshooting#try-the-latest-stable-version-of-npm). `npm@1.2.28` predates several important changes to how the npm client talks to its registry, including among other things the ability to connect securely via HTTPS. The warnings in your gist are a consequence of `npm@1.2.28` being so old (over 2 years at this point), and like the missing fields warnings I mentioned early, can be safely ignored. The errors in the gist you uploaded for @KenanY are due to race conditions that we finally fixed in `npm@2.1.5` and up.

**Comment by DaveLindberg** [comment]
Perfect! The sudo upgrade solved all issues; once complete, grunt and bower then installed perfectly. Thank you @KenanY  and @othiym23.
